src/workflow/ArcanistLandWorkflow.php

```diff
@@ -942,6 +942,82 @@ private function findRevision() {
         pht('Accepted'));
     }
 
+    // Check if all paths were reviewed by reviewers listed on METADATA files.
+    // If this check throws an exception - silently pass.
+    try {
+      $uber_metadata_unreviewed_paths = $this->getConduit()->callMethodSynchronous(
+        'uber_metadata.unreviewed_paths',
+        array(
+          'revisionid' => $rev_id,
+        ));
+      if (array_key_exists('pass', $uber_metadata_unreviewed_paths)
+        && false === $uber_metadata_unreviewed_paths['pass']
+      ) {
+
+        id(new PhutilConsoleWarning('WARNING', pht(
+          'Revision contains paths which were not reviewed by METADATA '.
+          'reviewers. Likely land operation will be blocked.'
+        )))->draw();
+
+        foreach ($uber_metadata_unreviewed_paths['groups'] as $group) {
+
+          $block = new PhutilConsoleBlock();
+          $reviewers = [];
+          if (isset($group['reviewers'][0])) {
+            if (!empty($group['reviewers'][0]['groups'])) {
```

Comment:
Could potentially be replaced with single `array_map(..., idx($group['reviewers'][0], "groups", array()))?`

Reply:
Potentially yes but that doesn't give any additional value.